larpandoracontent/LArControlFlow/TrackDirectionTool.h

```diff
@@ -0,0 +1,1089 @@
+/**
+ *  @file   larpandoracontent/LArControlFlow/TrackDirectionTool.h
+ *
+ *  @brief  Header file for the track direction finding tool AlgorithmTool class.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_TRACK_DIRECTION_TOOL_H
+#define LAR_TRACK_DIRECTION_TOOL_H 1
+
+#include "larpandoracontent/LArObjects/LArTwoDSlidingFitResult.h"
+#include "larpandoracontent/LArObjects/LArThreeDSlidingFitResult.h"
+#include "larpandoracontent/LArObjects/LArPfoObjects.h"
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "Pandora/AlgorithmTool.h"
+
+#include "larpandoracontent/LArControlFlow/MasterAlgorithm.h"
+
+#include <numeric>
+
+namespace lar_content
+{
+
+class TrackDirectionTool : public TrackDirectionBaseTool
+{
+
+friend void GetSplitChiSquared(int &npar, double *gin, double &f, double *par, int flag);
+
+public:
+
+    TrackDirectionTool();
+    pandora::StatusCode Initialize();
+
+    void FindDirections(const pandora::ParticleFlowObject *const pPfo, float &downProbability, const MasterAlgorithm *const pAlgorithm);
+
+    class HitCharge
+    {
+    public:
+
+        HitCharge(const pandora::CaloHit* caloHit, float &longitudinalPosition, float &hitWidth, float &hitCharge, float &uncertainty);
+        HitCharge();
+
+        const pandora::CaloHit* GetCaloHit() const;
+        float GetLongitudinalPosition() const;
+        float GetHitWidth() const;
+        float GetCharge() const;
+        float GetChargeOverWidth() const;
+        float GetUncertainty() const;
+
+        void SetDistanceToNN(float &distance);
+        float GetDistanceToNN() const;
+
+        void SetForwardsFitCharge(float &Q_fit_f); 
+        void SetForwardsSigma(float &f_sigma);
+        void SetForwardsDelta(float &forwardsDelta);
+        void SetForwardsChiSquared(float &forwardsHitChisquared);
+
+        void SetBackwardsFitCharge(float &Q_fit_b); 
+        void SetBackwardsSigma(float &b_sigma);
+        void SetBackwardsDelta(float &backwardsDelta);
+        void SetBackwardsChiSquared(float &backwardsHitChisquared);
+
+        float GetForwardsFitCharge(); 
+        float GetForwardsSigma();
+        float GetForwardsDelta();
+        float GetForwardsChiSquared();
+
+        float GetBackwardsFitCharge(); 
+        float GetBackwardsSigma();
+        float GetBackwardsDelta();
+        float GetBackwardsChiSquared();
+
+        bool                                       m_intails;
```

Comment:
Tried both, it does not take kindly to it. 
![Screenshot from 2021-04-09 11-00-26](https://user-images.githubusercontent.com/50299856/114164425-31013580-9923-11eb-891a-a2477242d065.png)

Reply:
Hi, This suggestion would indeed require additional changes. For instance, can make the member private and have a public accessor method to be able to return the information (ideally in a const manner, so that the information is returned, but other classes can't change the members of the TrackDirection::HitCharge class).

(Can add Get and Set functions if needs must, as that helps to track exactly how and where the public member variable is being manipulated. Ideally the member is set at construction and then never changes, so can then add a simple const member function to return the value. Always gets more complicated with more complicated usage!)

Reply:
Okay, done, thanks!